src/cargo/core/compiler/timings.rs

```diff
@@ -780,7 +780,15 @@ static HTML_CANVAS: &str = r#"
   </tr>
   <tr>
     <td><input type="range" min="0" max="30" step="0.1" value="0" id="min-unit-time"></td>
-    <td><input type="range" min="1" max="50" value="20" id="scale"></td>
+    <!--
+        The scale has the following interpretation:
+        - Value=N corresponds to a graph width of 200 + N*20 px.
```

Comment:
This removes a useful characteristic of the current implementation: it is possible to set two different timings charts to identical scales. This is something I do a lot to evaluate attempted build time improvements.

Certainly there is value in making it a good size by default, but I would like to request that the scale _value_ continues to mean the same “length per time” unit for all charts, and not be relative to the overall build time, so that it is possible to set identical scales on two different charts.

Reply:
See [#t-cargo > Making timings chart scalable](https://rust-lang.zulipchat.com/#narrow/channel/246057-t-cargo/topic/Making.20timings.20chart.20scalable/with/530629537)

From @Kobzol:

> ok, I see. so we would automatically set the scale value to match the viewport width, and adjust the maximum scale value so that you can always scale the graph up to 4096 pixels, regardless of how long the build took